ElectricityBillApi/Models/ElectricProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace ElectricityBillApi.Models
+{
+    public class ElectricProvider
+    {
+        private PowerPlant _subscibedPowerPlant;
```

Comment:
readonly probably is more suitable.

Reply:
If I make the field readonly I cannot use my subscription logic...